src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -0,0 +1,186 @@
+                                                                                                       helpLinkUri: null,
+                                                                                                       customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDuplicatedValue, RuleDuplicatedBitwiseValuePart);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSymbolStartAction(ssac =>
+            {
+                var enumSymbol = (INamedTypeSymbol)ssac.Symbol;
+                if (enumSymbol.TypeKind != TypeKind.Enum)
+                {
+                    return;
+                }
+
+                var invalidMembersByValue = new ConcurrentDictionary<object, ConcurrentBag<(SyntaxNode fieldSyntax, string fieldName)>>();
+
+                // Workaround to get the value of all enum fields that don't have an explicit initializer.
+                // We will start with all of the enum fields and remove the ones with an explicit initializer.
+                // See https://github.com/dotnet/roslyn/issues/40811
+                var filteredFields = enumSymbol.GetMembers()
+                    .OfType<IFieldSymbol>()
+                    .Where(f => !f.IsImplicitlyDeclared)
+                    .Select(x => new KeyValuePair<IFieldSymbol, object>(x, x.ConstantValue));
+                var enumFieldsWithImplicitValue = new ConcurrentDictionary<IFieldSymbol, object>(filteredFields);
+
+                // Collect duplicated values...
+                ssac.RegisterOperationAction(oc =>
+                {
+                    var fieldInitializerOperation = (IFieldInitializerOperation)oc.Operation;
+
+                    if (fieldInitializerOperation.InitializedFields.Length != 1 ||
+                        !fieldInitializerOperation.Value.ConstantValue.HasValue)
+                    {
+                        return;
+                    }
+
+                    var currentField = fieldInitializerOperation.InitializedFields[0];
+
+                    // Remove the explicitly initialized field
+                    enumFieldsWithImplicitValue.TryRemove(currentField, out _);
+
+                    var onlyReferencesOneField = GetFilteredDescendants(fieldInitializerOperation, op => op.Kind != OperationKind.Binary)
+                        .OfType<IFieldReferenceOperation>()
+                        .Where(fro => fro.Field.ContainingType.Equals(enumSymbol))
+                        .Any();
+                    if (onlyReferencesOneField)
+                    {
+                        return;
+                    }
+
+                    invalidMembersByValue.AddOrUpdate(fieldInitializerOperation.Value.ConstantValue.Value,
```

Comment:
I wonder if `invalidMembersByValue` is the correct name as you seem to be adding all fields which either have no RHS or whose RHS references 0 or greater then 1 fields. Should be called just `membersByValue` instead?